core/src/logging/filter/filter_builders.rs

```diff
@@ -0,0 +1,73 @@
+use crate::log_id::LogLevel;
+
+use super::{AddonFilter, FilterConfig, LogIdAddonFilter, LogIdModuleFilter};
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct FilterConfigBuilder {
+    logging_enabled: bool,
+    general_level: LogLevel,
+    general_addons: Vec<AddonFilter>,
+    /// LogIds set with `on[LogId]`
+    allowed_global_ids: Vec<LogIdAddonFilter>,
+    allowed_modules: Vec<LogIdModuleFilter>,
+}
+
+impl FilterConfigBuilder {
+    /// Creates a new builder for [`FilterConfig`].
+    pub fn new(level: LogLevel) -> Self {
+        Self {
+            general_level: level,
+            ..Default::default()
+        }
+    }
+
+    /// Disable logging in general.
+    pub fn disabled_logging(mut self) -> Self {
```

Comment:
minor nitpick. I think `disable_general_logging` sounds better, because it is a function.

See comment in filter.rs why `general`

Reply:
I still think that `disabled_general_logging` would fit better. This should be read as: build `FilterConfig` with `allowed_addons(...)`, `general_ids(...)`, `disabled_general_logging()` etc.

Reply:
you are right